### PR TITLE
Deny `derive_more` `error` feature and replace it with `thiserror`

### PR DESCRIPTION
**Pull request by bushrat011899**
# Objective

- Remove `derive_more`'s error derivation and replace it with `thiserror`

## Solution

- Added `derive_more`'s `error` feature to `deny.toml` to prevent it sneaking back in.
- Reverted to `thiserror` error derivation

## Notes

Merge conflicts were too numerous to revert the individual changes, so this reversion was done manually. Please scrutinise carefully during review.

### CODE REVIEW

File: crates/bevy_ecs/src/system/system_name.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use crate::{
 };
 use alloc::borrow::Cow;
 use core::ops::Deref;
-use derive_more::derive::{AsRef, Display, Into};
+use derive_more::derive::{AsRef, Into};
```
Comment (bushrat011899) [inline crates/bevy_ecs/src/system/system_name.rs:9]:
```suggestion
use derive_more::derive::{AsRef, Display, Into};
```

File: crates/bevy_ecs/src/system/system_name.rs [hunk 2]
```diff
@@ -34,7 +34,7 @@ use derive_more::derive::{AsRef, Display, Into};
 ///     logger.log("Hello");
 /// }
 /// ```
-#[derive(Debug, Into, Display, AsRef)]
+#[derive(Debug, Into, AsRef)]
```
Comment (bushrat011899) [inline crates/bevy_ecs/src/system/system_name.rs:37]:
```suggestion
#[derive(Debug, Into, Display, AsRef)]
```

File: crates/bevy_ecs/Cargo.toml [hunk 3]
```diff
@@ -32,10 +32,9 @@ concurrent-queue = "2.5.0"
 disqualified = "1.0"
 fixedbitset = "0.5"
 serde = { version = "1", optional = true, default-features = false }
+thiserror = { version = "2", default-features = false }
 derive_more = { version = "1", default-features = false, features = [
-  "error",
   "from",
```
Comment (bushrat011899) [inline crates/bevy_ecs/Cargo.toml:37]:
```suggestion
  "from",
  "display",
```

**Review by BenjaminBrienen — APPROVED** [review]
My eyes feel dizzy now lol

**Comment by alice-i-cecile** [comment]
Github's web interface hates this many files being changed, but I checked out the changes and they look fine. Merging ASAP to avoid merge conflicts.

**Pull request merged**